lib/supabase-auth-context.tsx

```
@@ -1,14 +1,15 @@
 import { createClient as createJsClient, type SupabaseClient } from "@supabase/supabase-js"
 import { useSession } from "@clerk/nextjs"
+import { BYPASS_CLERK } from "./dev-auth"
 import type { Database } from "./database.types"
 
 // Type for the context value, allowing null when not ready
 type SupabaseContextType = SupabaseClient<Database> | null;
 const SupabaseClientContext = createContext<SupabaseContextType>(null)
 
 export function SupabaseAuthProvider({ children }: { children: React.ReactNode }) {
-  const { session, isLoaded: isClerkLoaded } = useSession()
+  const { session, isLoaded: isClerkLoaded } = BYPASS_CLERK ? { session: null, isLoaded: true } : useSession()
```

ESLint

[error] React Hook "useSession" is called conditionally. React Hooks must be called in the exact same order in every component render.

(react-hooks/rules-of-hooks)